File: web/src/models/SubmissionReviewModel.ts
import type { ExhibitDescriptionModel } from './ExhibitDescriptionModel';
import type { SubmissionTicketModel } from './ExhibitSubmissionModel';

export type SubmissionStatus = 'Pending' | 'Accepted' | 'Rejected';

export interface SubmissionFile {
  id: string;
  originalFileName: string;
  storedFileName: string;
  viewUrl: string;
  downloadUrl: string;
  contentType: string;
  fileSize: number;
  storageProvider: string;
  status?: string;
  markedValue?: string | null;
  markedAt?: string | null;
  enteredValue?: string | null;
  enteredAt?: string | null;
  // Append-only description entries, oldest → newest (CES-42).
  descriptions: ExhibitDescriptionModel[];
  evidenceSourceType?: string | null;
  deletedAt?: string | null;
}

export interface ExhibitMarkModel {
  markedValue: string;
}

export interface ExhibitEnterModel {
  enteredValue: string;
}

export interface ExhibitEvidenceSourceModel {
  evidenceSourceType: string;
}

export interface ExhibitHistoryEntry {
  fieldName: string;
  oldValue?: string | null;
  newValue?: string | null;
  // See ExhibitDescriptionModel — id is the stored link, changedBy the resolved email.
  changedByUserId?: number | null;
  changedBy?: string | null;
  changedAtUTC: string;
}

export interface SubmissionReviewModel {
  id: number;
  submissionDate?: string;
  courtDateTime: string;
  location: string;
  room: string;
  locationName: string;
  status: SubmissionStatus;
  statusChangedDate?: string | null;
  exhibitCount: number;
  tickets: SubmissionTicketModel[];
  files: SubmissionFile[];
}

export interface SubmissionListFilter {
  submissionDateFrom?: string;
  submissionDateTo?: string;
  fileNumberText?: string;
  accusedName?: string;
  status?: SubmissionStatus | '';
  page: number;
  pageSize: number;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
}

export interface SubmissionActionModel {
  submissionId: number;
}

File: api/CES.Business/Models/SubmissionReviewModel.cs
using System;
using System.Collections.Generic;

namespace CES.Business.Models
{
    public class SubmissionReviewModel
    {
        public int Id { get; set; }
        public DateTime? SubmissionDate { get; set; }
        public string CourtDateTime { get; set; } = string.Empty;
        public string Location { get; set; } = string.Empty;
        public string Room { get; set; } = string.Empty;
        public string Status { get; set; } = "Pending";
        public DateTime? StatusChangedDate { get; set; }
        public int ExhibitCount { get; set; }
        public List<SubmissionTicketModel> Tickets { get; set; } = new();
        public List<SubmissionFile> Files { get; set; } = new List<SubmissionFile>();
    }

    public class SubmissionFile
    {
        public Guid Id { get; set; }
        public string OriginalFileName { get; set; } = string.Empty;
        public string StoredFileName { get; set; } = string.Empty;
        public string ContentType { get; set; } = string.Empty;
        public string Url { get; set; } = string.Empty;
        public long FileSize { get; set; }
        public string StorageProvider { get; set; } = string.Empty;
        public string Status { get; set; } = "Unclassified";
        public string? MarkedValue { get; set; }
        public DateTime? MarkedAt { get; set; }
        public string? EnteredValue { get; set; }
        public DateTime? EnteredAt { get; set; }
        // Append-only description entries, oldest → newest (CES-42).
        public List<ExhibitDescriptionEntryModel> Descriptions { get; set; } = new();
        public string? EvidenceSourceType { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}
